File: problems/3386-button-with-longest-push-time/solution.py
class Solution:
    def buttonWithLongestTime(self, events: List[List[int]]) -> int:
        max_time = events[0][1]
        max_index = events[0][0]

        for i in range(1, len(events)):
            time_diff = events[i][1] - events[i-1][1]
            if time_diff > max_time:
                max_time = time_diff
                max_index = events[i][0]
            elif time_diff == max_time:
                max_index = min(max_index, events[i][0])

        return max_index

File: problems/3386-button-with-longest-push-time/solution.cpp
class Solution {
public:
    int buttonWithLongestTime(vector<vector<int>>& events) {
        int maxTime = events[0][1];
        int buttonIndex = events[0][0];

        for (int i = 1; i < events.size(); ++i) {
            int timeDiff = events[i][1] - events[i - 1][1];
            if (timeDiff > maxTime) {
                maxTime = timeDiff;
                buttonIndex = events[i][0];
            }
        }

        return buttonIndex;
    }
};